packages/components/button/src/Button.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { ElementType, HTMLProps } from 'react';
+import { cx } from 'emotion';
+import { Flex, CommonProps } from '@contentful/f36-core';
+import { ChevronDown } from '@contentful/f36-icons';
+import { Icon, IconComponent } from '@contentful/f36-icon';
+
+import type { ButtonVariant, ButtonSize } from './types';
+import { styles } from './styles';
+import { Spinner } from '@contentful/f36-spinner';
+
+export interface ButtonProps
+  extends Omit<HTMLProps<HTMLButtonElement & HTMLAnchorElement>, 'size'>,
+    CommonProps {
+  children?: React.ReactNode;
+  /**
+   * Determines style variation of Button component
+   * @default secondary
+   */
+  variant?: ButtonVariant;
+  /**
+   * Determines size variation of Button component
+   * @default medium
+   */
+  size?: ButtonSize;
+  /**
+   * Applies active styles
+   * @default false
+   */
+  isActive?: boolean;
+  /**
+   * Disabled interaction and applies disabled styles
+   * @default false
+   */
+  disabled?: boolean;
+  /**
+   * Button html type attribute
+   */
+  type?: 'submit' | 'button' | 'reset';
+  /**
+   * Adds dropdown indicator icon
+   */
+  isDropdown?: boolean;
+  /**
+   * Expects any of the icon components
+   */
+  icon?: IconComponent;
+  /**
+   * Adds loading indicator icon and disables interactions
+   */
+  isLoading?: boolean;
+  /**
+   * Forces button to take 100% of the container
+   */
+  isFullWidth?: boolean;
+}
+
+const _Button = (props: ButtonProps, ref) => {
+  const {
+    children,
+    className,
+    testId,
+    variant = 'secondary',
+    size = 'medium',
+    href,
+    disabled,
+    type = !href ? 'button' : undefined,
+    icon,
+    isActive,
+    isDropdown,
+    isLoading,
+    isFullWidth,
+    ...otherProps
+  } = props;
+
+  const rootClassNames = cx(
+    className,
+    styles.button({
+      variant,
+      size,
+      isActive,
+      isDisabled: disabled,
+      isFullWidth,
+    }),
+  );
+
+  const Element: ElementType = href ? 'a' : 'button';
+
+  return (
+    <Element
+      ref={ref}
+      type={type}
+      className={rootClassNames}
+      href={href}
+      disabled={disabled}
+      data-test-id={testId}
```

Comment:
Sorry for nitpicking here but I think it would be nice to use `Primitive` here if we're not using `Box`, so that the test id prop etc. all work the same across all components.

Reply:
All good, that was my first [approach](https://github.com/contentful/forma-36/pull/938/commits/fb55b1af609ff9ad91277c89a6cfbe459f88d888#diff-6c783a3c20a6a5973aeb9a6d60e5308ad35ff07bf2e1d866b7d3f272151af9beR43), but unfortunately couldn't get types to work properly the current approach seemed more straight forward, If you want to pair on this, it would be great